src/Compilers/CSharp/Portable/Symbols/Tuples/TupleFieldSymbol.cs

```diff
@@ -140,25 +140,51 @@ public override sealed bool Equals(object obj)
     }
 
     /// <summary>
-    /// Represents an element field of a tuple type (such as (int, byte).Item1)
-    /// that is backed by a real field with the same name within the tuple underlying type.
+    /// Represents a non-element field of a tuple type (such as (int, byte).Rest)
+    /// that is backed by a real field within the tuple underlying type.
     /// </summary>
-    internal class TupleElementFieldSymbol : TupleFieldSymbol
+    internal sealed class TupleNonElementFieldSymbol : TupleFieldSymbol
+    {
+        public TupleNonElementFieldSymbol(TupleTypeSymbol container, FieldSymbol underlyingField, int tupleElementIndex)
+            : base(container, underlyingField, tupleElementIndex)
+        {
+        }
+
+        public sealed override int GetHashCode()
+        {
+            return GetHashCodeInternal();
+        }
+
+        public sealed override bool Equals(object obj)
+        {
+            if (!(obj is TupleNonElementFieldSymbol))
```

Comment:
I do not think the typecheck is needed. 
Distinct elements intentionally have distinct indices, so that comparing indices would be enough.